packages/console/src/pages/RoleDetails/RoleApplications/index.tsx

```diff
@@ -0,0 +1,193 @@
+import { useState } from 'react';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import { useOutletContext } from 'react-router-dom';
+import useSWR from 'swr';
+
+import Delete from '@/assets/icons/delete.svg';
+import MachineToMachineIconDark from '@/assets/icons/m2m-role-dark.svg';
+import MachineToMachineIcon from '@/assets/icons/m2m-role.svg';
+import Plus from '@/assets/icons/plus.svg';
+import EmptyDataPlaceholder from '@/components/EmptyDataPlaceholder';
+import ItemPreview from '@/components/ItemPreview';
+import { defaultPageSize } from '@/consts';
+import Button from '@/ds-components/Button';
+import ConfirmModal from '@/ds-components/ConfirmModal';
+import IconButton from '@/ds-components/IconButton';
+import Search from '@/ds-components/Search';
+import Table from '@/ds-components/Table';
+import { Tooltip } from '@/ds-components/Tip';
+import type { RequestError } from '@/hooks/use-api';
+import useApi from '@/hooks/use-api';
+import useSearchParametersWatcher from '@/hooks/use-search-parameters-watcher';
+import useTheme from '@/hooks/use-theme';
+import AssignRoleModal from '@/pages/Roles/components/AssignRoleModal';
+import { buildUrl, formatSearchKeyword } from '@/utils/url';
+
+import type { RoleDetailsOutletContext } from '../types';
+
+import * as styles from './index.module.scss';
+
+const pageSize = defaultPageSize;
+
+function RoleApplications() {
+  const {
+    role: { id: roleId },
+  } = useOutletContext<RoleDetailsOutletContext>();
+  const theme = useTheme();
+
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const [{ page, keyword }, updateSearchParameters] = useSearchParametersWatcher({
+    page: 1,
+    keyword: '',
+  });
+
+  const { data, error, mutate } = useSWR<[Application[], number], RequestError>(
+    roleId &&
+      buildUrl(`api/roles/${roleId}/applications`, {
+        page: String(page),
+        page_size: String(pageSize),
+        ...conditional(keyword && { search: formatSearchKeyword(keyword) }),
+      })
+  );
+
+  const isLoading = !data && !error;
+
+  const [applications, totalCount] = data ?? [];
+
+  const [isAssignModalOpen, setIsAssignModalOpen] = useState(false);
+  const [applicationToBeDeleted, setApplicationToBeDeleted] = useState<Application>();
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  const api = useApi();
+
+  const handleDelete = async () => {
+    if (!applicationToBeDeleted || isDeleting) {
+      return;
+    }
+    setIsDeleting(true);
+
+    try {
+      await api.delete(`api/roles/${roleId}/applications/${applicationToBeDeleted.id}`);
+      toast.success(t('role_details.applications.deleted', { name: applicationToBeDeleted.name }));
+      await mutate();
+      setApplicationToBeDeleted(undefined);
+    } finally {
+      setIsDeleting(false);
+    }
+  };
+
+  return (
+    <>
+      <Table
+        className={styles.usersTable}
```

Comment:
usersTable

Reply:
If it takes extra effort to design a shared component, could define a shared style class first.

Reply:
Better create a trackable issue to clean up these similar role assignment components.

Reply:
This refactor has already been put on TODO list, see LOG-7102.